bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/Activator.java

```diff
@@ -32,4 +54,33 @@ public void stop(BundleContext context) throws Exception {
     public static BundleContext getContext() {
         return Activator.context;
     }
+
+    /**
+     * Returns the {@link ThingTypeRegistry}
+     *
+     * @return {@link ThingTypeRegistry} or null
+     */
+    public static ThingTypeRegistry getThingTypeRegistry() {
+        return Activator.thingTypeRegistry;
+    }
+
+    private ServiceBinder channelTypeServiceBinder;
+
+    private ServiceBinder thingTypeServiceBinder;
+
+    @Override
+    public void start(BundleContext context) throws Exception {
+        Activator.context = context;
+        thingTypeServiceBinder = new ServiceBinder(context, new ThingTypeRegistryBinder());
+        thingTypeServiceBinder.open();
+        channelTypeServiceBinder = new ServiceBinder(context, new ChannelTypeRegistryBinder());
+        channelTypeServiceBinder.open();
+    }
+
+    @Override
+    public void stop(BundleContext context) throws Exception {
+        channelTypeServiceBinder.close();
```

Comment:
I´m not sure when a OSGI frameworks releases a reference to an Activator or if it might even recycle the old instance of the Activator (is this part of the spec?). But whatever it does, it has no effect on internal variables. Of course if you use services of another bundle, you have to free them manually. But just holding an internal objects can not make any difference (unless you don´t hold thousands of objects, which really free some memory in that case). Or am I wrong?